### PR TITLE
Add callback option to command line

### PR DESCRIPTION
**Pull request by loysollivier**
The callback option takes on argument. It should be the name of the file
located in ./templates/callback/ that contains all the "secret"
information such as the callback FQDN, lab name and token.

Signed-off-by: Loys Ollivier <lollivier@baylibre.com>

### CODE REVIEW

File: utils/agljobtemplate.py [hunk 1]
```diff
@@ -58,13 +73,21 @@ def render_job(self, url, machine, job_name="AGL-short-smoke", priority="medium"
...
-        job['test_templates'] = test_templates
-        job['rootfs_type'] = rfs_type
+        job = {
+            'name': job_name,
+            'yocto_machine': machine,
+            'priority': priority,
+            'urlbase': url,
+            'test_templates': test_templates,
+            'rootfs_type': rfs_type
+        }
```
Comment (jbrun3t) [inline utils/agljobtemplate.py]: Nicer way of putting things :+1:

File: utils/agljobtemplate.py [hunk 2]
```diff
@@ -58,13 +73,21 @@ def render_job(self, url, machine, job_name="AGL-short-smoke", priority="medium"
...
-        job['urlbase'] = url
-        job['test_templates'] = test_templates
-        job['rootfs_type'] = rfs_type
+        job = {
+            'name': job_name,
+            'yocto_machine': machine,
+            'priority': priority,
+            'urlbase': url,
+            'test_templates': test_templates,
+            'rootfs_type': rfs_type
```
Comment (jbrun3t) [inline utils/agljobtemplate.py]: This is going to conflict with #4

File: utils/create-jobs.py [hunk 3]
```diff
@@ -43,14 +45,14 @@ def main():
 
     if args.job_id is not None:
         args.urlbase = urlparse.urljoin(args.urlbase, args.job_id + '/')
-        args.job_name += ' - {}'.format(args.job_id)
+        args.job_name += '-{}'.format(args.job_id)
```
Comment (jbrun3t) [inline utils/create-jobs.py]: This and the following changes look like coding style correction which is good. It would be better to put them in another patch though

File: templates/base/agl-simple.jinja2 [hunk 4]
```diff
@@ -1,4 +1,4 @@
-{%- extends 'base/agl-base.jinja2' %}
+{%- extends 'base/agl-callback.jinja2' %}
 {%- block deploy -%}
```
Comment (jbrun3t) [inline templates/base/agl-simple.jinja2]: Not too sure about this.
I think it would be better to use to 'include' agl-callback.jinja2 in agl-base.jinja is callback_name is defined

**Review by jbrun3t — CHANGES_REQUESTED** [review]
Overall, I like this.
I would prefer if you could make a v2 with:
1) put coding style correction in a dedicated patch
2) use include instead of extend for the callback jinja

**Comment by loysollivier** [comment]
Agree to @jeromebrunet suggestions, you are the jinja2 master ;) 
I also removed all the coding style changes. I'll wait for this PR and #4 to be merged. On top of that I'll make a PR just about coding style

**Comment by khilman** [comment]
Can you  rebase this onto the master branch please?
